src/INode.java

```diff
@@ -189,9 +189,29 @@ private ASTValue evalAnd(WordsEnvironment environment) {
 		throw new AssertionError("Not yet implemented");
 	}
 
-	private ASTValue evalAssign(WordsEnvironment environment) {
-		// TODO
-		throw new AssertionError("Not yet implemented");	
+	private ASTValue evalAssign(WordsEnvironment environment) throws WordsRuntimeException {
+		WordsObject obj = children.get(0).eval(environment).objValue;
+		String propertyName = children.get(1).eval(environment).stringValue;
+		ASTValue propertyASTValue = children.get(2).eval(environment);
+		
+		WordsProperty wordsProp = null;
+		switch (propertyASTValue.type) {
+			case NUM:
+				wordsProp = new WordsProperty(propertyASTValue.numValue);
+				break;
+			case STRING:
+				wordsProp = new WordsProperty(propertyASTValue.stringValue);
+				break;
+			case OBJ:
+				wordsProp = new WordsProperty(propertyASTValue.objValue);
+				break;
+			default:
+				wordsProp = new WordsProperty(WordsProperty.PropertyType.NOTHING);
+		}
+	
+		obj.setProperty(propertyName, wordsProp);
```

Comment:
Can you fix set property so it throws the exception it's supposed to (just took a look in WordsObject.setProperty and it has a to do. Figure that this branch is a good place to fix that)?